sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerBuilder.java

```diff
@@ -76,9 +77,14 @@
         this(
                 serviceSpec,
                 schedulerConfig,
-                schedulerConfig.isStateCacheEnabled() ?
-                        new PersisterCache(CuratorPersister.newBuilder(serviceSpec).build()) :
-                        CuratorPersister.newBuilder(serviceSpec).build());
+                maybeWrapPersister(schedulerConfig, CuratorPersister.newBuilder(serviceSpec).build()));
+    }
+
+    private static Persister maybeWrapPersister(
+            SchedulerConfig schedulerConfig, final CuratorPersister curatorPersister) throws PersisterException {
+        // TODO(mowsiany): make the log wrapping optional
```

Comment:
For reference, this option could be env-driven and provided by `schedulerConfig`